pkgs/applications/editors/vim/plugins/overrides.nix

```diff
@@ -2129,6 +2129,40 @@ in
     dependencies = [ self.nvim-treesitter ];
   };
 
+  nvim-java = super.nvim-java.overrideAttrs {
+    dependencies = with self; [
+      lua-async-await
+      mason-nvim
+      nui-nvim
+      nvim-dap
+      nvim-java-core
+      nvim-java-dap
+      nvim-java-refactor
+      nvim-java-test
+      nvim-lspconfig
+    ];
+  };
+
+  nvim-java-core = super.nvim-java-core.overrideAttrs {
+    dependencies = with self; [
+      mason-nvim
+      nvim-lspconfig
+    ];
+  };
+
+  nvim-java-dap = super.nvim-java-dap.overrideAttrs {
+    dependencies = [ self.nvim-java-core ];
+  };
+
+  nvim-java-refactor = super.nvim-java-refactor.overrideAttrs {
+    dependencies = [ self.nvim-java-core ];
+    nvimSkipModule = [
+      # Requires some mysterious `java.utils.ui` module
```

Comment:
Hmm interesting.. can't find anything easily on it.

Reply:
Huh? https://github.com/nvim-java/nvim-java/blob/main/lua/java/utils/ui.lua